omod/src/main/java/org/openmrs/module/ipdapp/fragment/controller/PatientInfoFragmentController.java

```diff
@@ -188,101 +187,101 @@ public void dischargePatient( @RequestParam(value ="dischargeAdmittedID", requir
+            Set<Obs> obses = new HashSet(ipdEncounter.getAllObs());
 
-        ipdEncounter.setObs(null);
+            ipdEncounter.setObs(null);
 
-        List<Concept> listConceptDianosisOfIpdEncounter = new ArrayList<Concept>();
-        List<Concept> listConceptProcedureOfIpdEncounter = new ArrayList<Concept>();
-        if (CollectionUtils.isNotEmpty(listObsOfIpdEncounter)) {
-            for (Obs obx : obses) {
-                if (obx.getConcept().getConceptId().equals(cDiagnosis.getConceptId())) {
-                    listConceptDianosisOfIpdEncounter.add(obx.getValueCoded());
-                }
+            List<Concept> listConceptDianosisOfIpdEncounter = new ArrayList<Concept>();
+            List<Concept> listConceptProcedureOfIpdEncounter = new ArrayList<Concept>();
+            if (CollectionUtils.isNotEmpty(listObsOfIpdEncounter)) {
+                for (Obs obx : obses) {
+                    if (obx.getConcept().getConceptId().equals(cDiagnosis.getConceptId())) {
+                        listConceptDianosisOfIpdEncounter.add(obx.getValueCoded());
+                    }
 
-                if (obx.getConcept().getConceptId().equals( cProcedure.getConceptId())) {
-                    listConceptProcedureOfIpdEncounter.add(obx.getValueCoded());
+                    if (obx.getConcept().getConceptId().equals(cProcedure.getConceptId())) {
+                        listConceptProcedureOfIpdEncounter.add(obx.getValueCoded());
+                    }
                 }
             }
-        }
 
-        List<Concept> listConceptDiagnosis = new ArrayList<Concept>();
-
-        if(selectedDiagnosisList!=null){
-            for (Integer cId : selectedDiagnosisList) {
-                Concept cons = conceptService.getConcept(cId);
-                listConceptDiagnosis.add(cons);
-                //if (!listConceptDianosisOfIpdEncounter.contains(cons)) {
-                Obs obsDiagnosis = new Obs();
-                //obsDiagnosis.setObsGroup(obsGroup);
-                obsDiagnosis.setConcept(cDiagnosis);
-                obsDiagnosis.setValueCoded(cons);
-                obsDiagnosis.setCreator(user);
-                obsDiagnosis.setObsDatetime(date);
-                obsDiagnosis.setLocation(location);
-                obsDiagnosis.setDateCreated(date);
-                obsDiagnosis.setPatient(ipdEncounter.getPatient());
-                obsDiagnosis.setEncounter(ipdEncounter);
-                obsDiagnosis = Context.getObsService().saveObs(obsDiagnosis, "update obs diagnosis if need");
-                obses.add(obsDiagnosis);
-                //}
+            List<Concept> listConceptDiagnosis = new ArrayList<Concept>();
+
+            if (selectedDiagnosisList != null) {
+                for (Integer cId : selectedDiagnosisList) {
+                    Concept cons = conceptService.getConcept(cId);
+                    listConceptDiagnosis.add(cons);
+                    //if (!listConceptDianosisOfIpdEncounter.contains(cons)) {
```

Comment:
@franqq please remove the commented out codes, if we really need them then uncomment them